src/hw4_exceptions/src/main/java/com/company/hw/User.java

```diff
@@ -0,0 +1,180 @@
+            }
+        }
+    }
+
+    //2
+    void getFilteredByAuthor(String author) throws NotAnyBooks{
+        List<Library> tempList = new ArrayList<>();
+        for (Library element : books) {
+            if (element.available && element.getAuthor().contains(author)) {
+                tempList.add(element);
+            }
+        }
+        System.out.println(tempList);
+        if (tempList.isEmpty()) {
+            throw new NotAnyBooks("I throw exception");
+        }
+    }
+
+    //3
+    void sortAlpabetically() {
+        books.sort(new SortByBook());
+        for (Library element : books) {
+            System.out.println(element);
+        }
+    }
+
+    //4
+    void hasEarlierBooks(int year) throws NotAnyBooks{
+        for (Library next : books) {
+            if (year < 1604) {
+                throw new NotAnyBooks("I throw exception");
+            }
+            else if(next.getYear() < year){
+                System.out.println(next);
+            }
+        }
+    }
+
+    //5
+    void checkGenre(String genre) throws NotAnyBooks{
+        List<Library> tempList = new ArrayList<>();
+        for (Library next : books) {
+            if (next.getClass().toString().contains(genre)) {
+                tempList.add(next);
+            }
+        }
+        System.out.println(tempList);
+
+        if (tempList.isEmpty()) {
+            throw new NotAnyBooks("I throw exception");
+        }
+
+    }
+
+    //6
+    boolean hasNotBooksByAuthor(String author) {
+        List<Library> tempList = new ArrayList<>();
+        for (Library next : books) {
+            if (!next.getAuthor().contains(author)) {
+                tempList.add(next);
+            }
+        }
+        return books.size() == tempList.size();
+    }
+
+    //7
+    void filteredByGenre() throws NotAnyBooks {
+        List<Library> listOfNovel = new ArrayList<>();
+        List<Library> listOfComics = new ArrayList<>();
+        List<Library> listOfEncyclopedia = new ArrayList<>();
+        List<Library> tempList = new ArrayList<>();
+        for (Library element : books) {
+            if (element.getClass().equals(novelExample.getClass())) {
+                listOfNovel.add(element);
+                listOfNovel.sort(new SortByYear());
+
+            } else if (element.getClass().equals(comicsExample.getClass())) {
+                listOfComics.add(element);
+                listOfComics.sort(new SortByYear());
+
+            } else if (element.getClass().equals(encyclopediaExample.getClass())) {
+                listOfEncyclopedia.add(element);
+                listOfEncyclopedia.sort(new SortByYear());
+            }
+        }
+        try {
+            tempList.add(listOfNovel.get(0));
+            tempList.add(listOfComics.get(0));
+            tempList.add(listOfEncyclopedia.get(0));
+            tempList.sort(new SortByBook());
+        }
+        catch(IndexOutOfBoundsException ex){
+            throw new NotAnyBooks("I throw exception");
```

Comment:
NotAnyBooks - is not convention name of Exception. NoSuchBookException or smthng similar

Reply:
fixed